mantidimaging/gui/windows/spectrum_viewer/model.py

```diff
@@ -481,3 +478,11 @@ def set_relevant_tof_units(self) -> None:
                     self.tof_data = self.units.tof_seconds_to_energy()
                 self.tof_plot_range = (self.tof_data.min(), self.tof_data.max())
                 self.tof_range = (0, self.tof_data.size)
+
+    def set_tof_unit_mode_for_stack(self):
+        if self.get_stack_time_of_flight() is None or self.tof_data is None:
+            self.tof_mode = ToFUnitMode.IMAGE_NUMBER
+        elif self.tof_mode is not ToFUnitMode.WAVELENGTH or self.tof_mode is not ToFUnitMode.IMAGE_NUMBER:
```

Comment:
Is the aim here to reset to `IMAGE_NUMBER` if there is not tof data (and not change if if there is tof data)?

Reply:
The idea here is if there is no spectrum file, then the mode should be `IMAGE_NUMBER`, if there is a spectrum file then the default mode should be `IMAGE_NUMBER`. 
However if you are in a different mode e.g. `ENERGY` and you want to switch to another data set which also has a spectrum file, then it plots the spectrum in the same mode so you dont have to keep switching the tof mode when looking at data.

It should be `and` rather than `or`, so if the mode isnt `IMAGE_NUMBER` or `WAVELENGTH` then the selected mode stays the same.

Good catch!

Reply:
and if it is `IMAGE_NUMBER` or `WAVELENGTH` it also stays the same